### PR TITLE
Tree by chromosome

### PR DESCRIPTION
**Pull request by joshuak94**
Create one interval tree per chromosome.

Currently a work in progress, don't merge yet!

### CODE REVIEW

File: include/bamit/IntervalNode.hpp [hunk 1]
```diff
@@ -341,22 +318,41 @@ void get_overlap_file_offset(std::unique_ptr<IntervalNode> const & node,
...
+    // Look for the file offset using the tree for the starting chromosome.
+    if (std::get<0>(start) == std::get<0>(end)) // Searching in one chromosome.
+    {
+        get_overlap_file_offset(node_list[std::get<0>(start)], std::get<1>(start), std::get<1>(end), offset_pos);
+    }
+    else // Searching across multiple chromosomes.
+    {
+        for (uint32_t i = std::get<0>(start); i < std::get<0>(end); ++i)
+        {
+            // Start at given start only for the first chromosome, otherwise start searching from 0.
```
Comment (MitraDarja) [inline include/bamit/IntervalNode.hpp:337]: Mabye add explaination why.

File: include/bamit/IntervalNode.hpp [hunk 2]
```diff
@@ -341,22 +318,41 @@ void get_overlap_file_offset(std::unique_ptr<IntervalNode> const & node,
...
+            uint32_t start_position = std::get<0>(start) == i ? std::get<1>(start) : 0;
+            // End at given end if at the last chromosome, otherwise end at end of chromosome.
+            uint32_t end_position = std::get<0>(end) == i ? std::get<1>(end) : std::numeric_limits<uint32_t>::max();
+            get_overlap_file_offset(node_list[i], start_position, end_position, offset_pos);
+
+            // If we find the left-most offset we can stop. Otherwise we have to check the next tree.
```
Comment (MitraDarja) [inline include/bamit/IntervalNode.hpp:343]: Add if the given interval spans more than one chromosome, right?

File: test/api/search_test.cpp [hunk 3]
```diff
@@ -8,20 +8,22 @@
...
     std::filesystem::path result_sam_path{tmp_dir/"result.sam"};
-    bamit::sam_file_input_type input_bam{input};
     bamit::Position start{1, 100};
     bamit::Position end{1, 110};
-    get_overlap_records(input_bam, root, start, end, result_sam_path);
+    // Potentially undefined behavior in our seqan branch: input_file is parsed in the tree construction,
+    // so using the same input_file variable here leads to incorrect results (probably because the iterator no longer
+    // points to the beginning of the file).
+    // Temporary fix: use new object.
+    // Permanent fix: seqan3 is changing how you get/use the file position.
```
Comment (MitraDarja) [inline test/api/search_test.cpp:24]: Is the test file the best place to mention this problem? Should be in the include file and maybe add a #TODO or something?

**Review by MitraDarja — CHANGES_REQUESTED** [review]
LGTM, just some minor things about the docu and please fix the doc error from the CI. :)

**Comment by joshuak94** [comment]
@MitraDarja I made a few changes since the last time, can you look again.

**Review by MitraDarja — APPROVED** [review]

**Pull request merged**